File: gva/server/plugin/pmocker_team/router/team.go
package team

import (
	"github.com/flipped-aurora/gin-vue-admin/server/middleware"
	"github.com/gin-gonic/gin"
)

type Team struct{}

func (r *Team) Init(public, private *gin.RouterGroup) {
	// 写操作记录操作日志
	{
		g := private.Group("team").Use(middleware.OperationRecord())
		// 团队成员
		g.POST("member/create", ApiGroupApp.CreateMember)
		g.DELETE("member/delete", ApiGroupApp.DeleteMember)
		g.PUT("member/update", ApiGroupApp.UpdateMember)
		g.POST("member/transition", ApiGroupApp.TransitionMember)
		// 角色定义
		g.POST("role/create", ApiGroupApp.CreateRole)
		g.DELETE("role/delete", ApiGroupApp.DeleteRole)
		g.PUT("role/update", ApiGroupApp.UpdateRole)
		g.POST("role/transition", ApiGroupApp.TransitionRole)
		// 培训记录
		g.POST("training/create", ApiGroupApp.CreateTraining)
		g.DELETE("training/delete", ApiGroupApp.DeleteTraining)
		g.PUT("training/update", ApiGroupApp.UpdateTraining)
		g.POST("training/transition", ApiGroupApp.TransitionTraining)
		// 绩效评估
		g.POST("performance/create", ApiGroupApp.CreatePerformance)
		g.DELETE("performance/delete", ApiGroupApp.DeletePerformance)
		g.PUT("performance/update", ApiGroupApp.UpdatePerformance)
		g.POST("performance/transition", ApiGroupApp.TransitionPerformance)
	}
	// 读操作不记录操作日志
	{
		g := private.Group("team")
		// 团队成员
		g.GET("member/find", ApiGroupApp.FindMember)
		g.GET("member/list", ApiGroupApp.ListMember)
		// 角色定义
		g.GET("role/find", ApiGroupApp.FindRole)
		g.GET("role/list", ApiGroupApp.ListRole)
		// 培训记录
		g.GET("training/find", ApiGroupApp.FindTraining)
		g.GET("training/list", ApiGroupApp.ListTraining)
		// 绩效评估
		g.GET("performance/find", ApiGroupApp.FindPerformance)
		g.GET("performance/list", ApiGroupApp.ListPerformance)
	}
}

File: gva/web/src/api/pmocker/team.js
import service from '@/utils/request'

// ---- 团队成员 ----
export const createMember = (data) => service({ url: '/pmocker/team/member/create', method: 'post', data })
export const deleteMember = (params) => service({ url: '/pmocker/team/member/delete', method: 'delete', params })
export const updateMember = (data) => service({ url: '/pmocker/team/member/update', method: 'put', data })
export const findMember = (params) => service({ url: '/pmocker/team/member/find', method: 'get', params })
export const listMember = (params) => service({ url: '/pmocker/team/member/list', method: 'get', params })
export const transitionMember = (data) => service({ url: '/pmocker/team/member/transition', method: 'post', data })

// ---- 角色定义 ----
export const createRole = (data) => service({ url: '/pmocker/team/role/create', method: 'post', data })
export const deleteRole = (params) => service({ url: '/pmocker/team/role/delete', method: 'delete', params })
export const updateRole = (data) => service({ url: '/pmocker/team/role/update', method: 'put', data })
export const findRole = (params) => service({ url: '/pmocker/team/role/find', method: 'get', params })
export const listRole = (params) => service({ url: '/pmocker/team/role/list', method: 'get', params })
export const transitionRole = (data) => service({ url: '/pmocker/team/role/transition', method: 'post', data })

// ---- 培训记录 ----
export const createTraining = (data) => service({ url: '/pmocker/team/training/create', method: 'post', data })
export const deleteTraining = (params) => service({ url: '/pmocker/team/training/delete', method: 'delete', params })
export const updateTraining = (data) => service({ url: '/pmocker/team/training/update', method: 'put', data })
export const findTraining = (params) => service({ url: '/pmocker/team/training/find', method: 'get', params })
export const listTraining = (params) => service({ url: '/pmocker/team/training/list', method: 'get', params })
export const transitionTraining = (data) => service({ url: '/pmocker/team/training/transition', method: 'post', data })

// ---- 绩效评估 ----
export const createPerformance = (data) => service({ url: '/pmocker/team/performance/create', method: 'post', data })
export const deletePerformance = (params) => service({ url: '/pmocker/team/performance/delete', method: 'delete', params })
export const updatePerformance = (data) => service({ url: '/pmocker/team/performance/update', method: 'put', data })
export const findPerformance = (params) => service({ url: '/pmocker/team/performance/find', method: 'get', params })
export const listPerformance = (params) => service({ url: '/pmocker/team/performance/list', method: 'get', params })
export const transitionPerformance = (data) => service({ url: '/pmocker/team/performance/transition', method: 'post', data })
